cmd/export_ledger_entry_changes.go

```diff
@@ -18,6 +25,90 @@ confirmed by the Stellar network.
 If no data type flags are set, then by default all of them are exported. If any are set, it is assumed that the others should not
 be exported.`,
 	Run: func(cmd *cobra.Command, args []string) {
+		startNum, endNum, _, _, _ := utils.MustBasicFlags(cmd.Flags(), cmdLogger)
+		execPath, configPath, exportAccounts, exportOffers, exportTrustlines, _ := utils.MustCoreFlags(cmd.Flags(), cmdLogger)
+
+		if configPath == "" && endNum == 0 {
+			cmdLogger.Fatal("A path to a config file for stellar-core is mandatory when performing an unbounded export")
+		}
+
+		var err error
+		execPath, err = filepath.Abs(execPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for stellar-core executable: ", err)
+		}
+
+		configPath, err = filepath.Abs(configPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for the config file: ", err)
+		}
+
+		/*var outFile *os.File
+		if !useStdout {
+			outFile = mustOutFile(path)
+		}*/
+
+		core, err := input.PrepareCaptiveCore(execPath, configPath, startNum, endNum)
+		if err != nil {
+			cmdLogger.Fatal("error creating a prepared captive core instance: ", err)
+		}
+
+		accChannel, offChannel, trustChannel := createChangeChannels(exportAccounts, exportOffers, exportTrustlines)
+		go input.StreamChanges(core, startNum, endNum, accChannel, offChannel, trustChannel)
+
+		transformedAccounts := make([]transform.AccountOutput, 0)
+		transformedOffers := make([]transform.OfferOutput, 0)
+		transformedTrustlines := make([]transform.TrustlineOutput, 0)
+
+		for {
+
+			select {
+			case entry, ok := <-accChannel:
+				if !ok {
+					accChannel = nil
+				} else {
+					acc, err := transform.TransformAccount(entry)
+					if err == nil {
+						transformedAccounts = append(transformedAccounts, acc)
+					} else {
+						cmdLogger.Error("error transforming account entry: ", err)
+					}
+				}
+
+			case entry, ok := <-offChannel:
+				wrappedEntry := ingestio.Change{Type: xdr.LedgerEntryTypeOffer, Post: &entry}
+				offer, err := transform.TransformOffer(wrappedEntry)
+				if !ok {
+					offChannel = nil
+				} else {
+					if err == nil {
+						transformedOffers = append(transformedOffers, offer)
+					} else {
+						cmdLogger.Error("error transforming offer entry: ", err)
+					}
+				}
+
+			case entry, ok := <-trustChannel:
+				trust, err := transform.TransformTrustline(entry)
+				if !ok {
+					trustChannel = nil
+				} else {
+					if err == nil {
+						transformedTrustlines = append(transformedTrustlines, trust)
+					} else {
+						cmdLogger.Error("error transforming trustline entry: ", err)
+					}
+				}
+			}
+
+			if accChannel == nil && offChannel == nil && trustChannel == nil {
+				break
+			}
+		}
+
+		fmt.Println(transformedAccounts)
```

Comment:
To be specific: I think this will work in a bounded setting, but when unbounded, we likely want to apply some batching strategy. Otherwise, I don't think we'd ever get printed output, but we would eventually get an out-of-memory error once these arrays of changes get too large.

Reply:
Yeah, this was just for testing bounded ranges. I should have put a TODO here for the next PR.